index.bs

```diff
@@ -5966,6 +6027,8 @@ partial interface MLGraphBuilder {
   <summary>
     The <dfn method for=MLGraphBuilder>where(|condition|, |input|, |other|)</dfn> method steps are:
   </summary>
+    1. If [=this=].{{MLGraphBuilder/[[hasBuilt]]}} is true, then [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
```

Comment:
🫤 Seeing "if hasBuilt is true then throw this kind of error" repeated in the logic of *every* single operator feels unclean. Can we have a common graph builder "operator initialization" function that they all call, and that error check can be the very first (and currently only) step in it, kinda like how we centralized [casting](https://www.w3.org/TR/webnn/#algorithms-casting) and broadcasting functionality? Mind you, every single operator still then needs "1. Perform builder-operator-initialization", but it feels 🧼➕ imo. Opinion?

Reply:
Hmm I agree the current state is annoyingly duplicative, but I couldn't decide on a useful name of a helper method... This check is performed in `MLGraphBuilder.build()` as well all methods which mint an `MLOperand`. Calling a "builder-operator-initialization" algorithm from the `build()` steps is a bit odd (and nothing is being initialized anyways).

Lacking a good name, I figured inlining the one-liner would be more readable ¯\\_(ツ)_/¯

If you can think of a better name then I'm happy to change it :P

Reply:
The pervasive naming challenge... Let me ponder a bit, after some sleep.

Reply:
Throwing out another idea: we could maybe use a [Bikeshed text macro](https://speced.github.io/bikeshed/#metadata-text-macro) to reduce the burden on spec authors. We [define one today for EMULATED](https://github.com/webmachinelearning/webnn/blob/fe96a7809e33fc7dd1cf58717743089f559d8e96/index.bs#L39) to avoid repetition. This doesn't reduce the burden on spec readers, though.

Reply:
True, though we'd still have to name the macro :P 

`THROW_IF_BUILT`?

Reply:
No bright insights came to me after sleep or over the weekend 😅. Going to merge now and maybe think of something later.